src/MongoDB/BulkWrite.c

```diff
@@ -73,6 +73,70 @@ static inline bool php_phongo_bulkwrite_update_has_operators(bson_t* bupdate) /*
 	return false;
 } /* }}} */
 
+/* Returns whether the BSON array's keys are a sequence of integer strings
+ * starting with "0". BSON_APPEND_ARRAY considers it the caller's responsibility
+ * to ensure that the array's keys are properly formatted. */
```

Comment:
And this is an equivalent to bson-encode.c's ``php_phongo_is_array_or_document``? Should they be placed together perhaps? I don't mind, so please do what makes sense for you.

Reply:
`php_phongo_is_array_or_document()` operates on a `zval`, while this validates a `bson_t`.

We can't work with the `zval` here, as we're not inspecting it until it's already been converted to a `bson_t`. In theory, this allows a Serializable value to produce a BSON array for this option.